airbyte_cdk/cli/airbyte_cdk/_manifest.py

```diff
@@ -19,6 +31,176 @@ def manifest_cli_group() -> None:
     pass
 
 
+@manifest_cli_group.command("validate")
+@click.option(
+    "--manifest-path",
+    type=click.Path(exists=True, path_type=Path),
+    default="manifest.yaml",
+    help="Path to the manifest file to validate (default: manifest.yaml)",
+)
+def validate_manifest(manifest_path: Path) -> None:
+    """Validate a manifest file against the declarative component schema.
+
+    This command validates the manifest file and checks version compatibility.
+    If validation fails, it will suggest running the migrate command if needed.
+
+    Exit codes:
+    \\b
+    0: Manifest is valid and up to date
+    1: Manifest is valid but needs migration, or general errors
+    2: Manifest has validation errors that are fixable via migration
+    3: Manifest has validation errors that are NOT fixable via migration
+    """
+    try:
+        with open(manifest_path, "r") as f:
+            manifest_dict = yaml.safe_load(f)
+
+        if not isinstance(manifest_dict, dict):
+            click.echo(
+                f"❌ Error: Manifest file {manifest_path} does not contain a valid YAML dictionary",
+                err=True,
+            )
+            sys.exit(1)
+
+        schema = _get_declarative_component_schema()
+
+        validation_error = None
+        try:
+            validate(manifest_dict, schema)
+            original_is_valid = True
+        except ValidationError as e:
+            original_is_valid = False
+            validation_error = e
+
+        migration_handler = ManifestMigrationHandler(copy.deepcopy(manifest_dict))
```

Comment:
Should we exit here if `original_is_valid`? Premise: No point in checking migration path when we're already valid at this point.